src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -384,6 +384,11 @@ protected SCMRevision retrieve(SCMHead head, TaskListener listener) throws IOExc
         List<? extends BitbucketBranch> branches = bitbucket.getBranches();
         for (BitbucketBranch b : branches) {
             if (b.getName().equals(bbHead.getBranchName())) {
+                // If the head of the revision is null we might be using version 3.x.x
+                if (b.getRawNode() == null) {
+                    listener.getLogger().format("Cannot resolve the hash of the revision in branch %s. Perhaps you are using a BitBucket Server or Cloud version minor than 4.x.x ", b.getName());
```

Comment:
You know that my english is very bad but maybe something like _Verify if the current version is compatible with the API of the Bitbucket Server instance or Bitbucket Cloud service._